app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx

```diff
@@ -115,17 +117,27 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
       key: "status",
       title: "Status",
       width: 120,
-      render: (rule: RuleTableDataType) => {
-        const checked = rule?.status === RuleObjStatus.ACTIVE ? true : false;
+      render: (_, rule: RuleTableDataType, index) => {
+        const isRuleActive = rule.status === RuleObjStatus.ACTIVE;
+
         return (
-          <Switch
-            size="small"
-            checked={checked}
-            onChange={(checked: boolean, e) => {
-              e.stopPropagation();
-              handleStatusToggle([rule], checked);
-            }}
-          />
+          <PremiumFeature
+            disabled={isRuleActive}
+            features={[FeatureLimitType.num_active_rules]}
+            popoverPlacement="left"
+            onContinue={() => handleStatusToggle([rule])}
+            source="rule_list_status_switch"
+          >
+            <Switch
+              size="small"
+              checked={isRuleActive}
+              data-tour-id={index === 0 ? "rule-table-switch-status" : null}
+              onChange={(checked: boolean, e) => {
+                e.stopPropagation();
+                handleStatusToggle([rule]);
+              }}
```

Comment:
Remove this onChange, `PremiumFeature` wrapper will handle it if toggle is allowed.